modules/client-akka/src/main/scala/be/wegenenverkeer/atomium/client/AbstractAtomiumFeedClientActor.scala

```diff
@@ -0,0 +1,237 @@
+                                                                              eventFormat: Format[EVENT],
+                                                                              feedEntryFormat: Reads[FEEDENTRY],
+                                                                              monitorActor: ActorRef
+                                                                               ) extends PersistentActor with ActorLogging {
+
+  import AbstractAtomiumFeedClientActor.Protocol._
+
+  lazy val atomiumClient = new japi.client.AtomiumClient.Builder()
+    .setBaseUrl(config.baseUrl)
+    .setAcceptJson()
+    .setConnectTimeout(config.connectTimeout.toMillis.toInt)
+    .build
+    .asScala
+
+  var positionInFeed: Option[FeedPosition] = None
+
+  def receiveRecover: Receive = {
+    case RecoveryCompleted =>
+      log.debug("{}: Recovery completed, start the feed observer", config.feedUrl)
+      subscribeToFeed()
+
+    case jsValue: JsValue => onJsValueEvent(jsValue)
+
+    case x: SnapshotOffer => positionInFeed = x.snapshot.asInstanceOf[Option[FeedPosition]]
+  }
+
+  def createEventToPublish(c: NewFeedEntryReceived[FEEDENTRY]): EVENT
+
+  def publishEvent(event: EVENT): Unit
+
+  def status(atomiumFeedClientStatus: AtomiumFeedClientStatus): Unit = ()
+
+  override def receiveCommand: Receive = {
+
+    case c: NewFeedEntryReceived[FEEDENTRY] =>
+      val event = createEventToPublish(c)
+      persistEvent(event) { evt =>
+        updateState(evt)
+        log.debug("{}: entry on position {} saved succesfully, putting on eventbus", config.feedUrl, evt.position)
+        publishEvent(event)
+
+        status(AtomiumFeedClientStatus(
+          ok = true,
+          status = "receiving",
+          feedPosition = Some(evt.position.selfHref),
+          entryId = Some(evt.position.entryId),
+          receivedOn = Some(evt.position.receivedOn)))
+      }
+
+    case e: FeedReadError =>
+      // the atomium client observer is now unsubscribed, so we will receive no more events in this subscription
+      log.error(e.error, "{}: Error while reading feed, last known position was {}, will stop now", config.feedUrl, positionInFeed)
+
+      // close should not cause errors here
+      Try(atomiumClient.close())
+
+      status(AtomiumFeedClientStatus(
+        ok = false,
+        status = "error",
+        error = Some(e.error.getMessage),
+        feedPosition = positionInFeed.map(_.selfHref),
+        entryId = positionInFeed.map(_.entryId),
+        receivedOn = positionInFeed.map(_.receivedOn)
+      ))
+
+      context.stop(self)
+
+    case GetStatusInfo =>
+      sender() ! positionInFeed
+  }
+
+  private def updateState(event: EVENT) = {
+    positionInFeed = Some(event.position)
+  }
+
+
+  protected def subscribeToFeed(): Unit = {
+
+    def createObservable: Observable[FeedEntry[JsonNode]] = {
+      val feedObservableBuilder = atomiumClient.feed(config.feedUrl, classOf[JsonNode])
```

Comment:
The class documentation points out that it is a generic actor that consumes Atomium feeds _(in json format)_. I could rename the class to AbstractAtomiumJsonFeedClientActor?